files/image_config/caclmgrd/caclmgrd

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python
+#
+# caclmgrd
+#
+# Control plane ACL manager daemon for SONiC
+#
+#  Upon starting, this daemon reads control plane ACL tables and rules from
+#  Config DB, converts the rules into iptables rules and installs the iptables
+#  rules. The daemon then indefintely listens for notifications from Config DB
+#  and updates iptables rules if control plane ACL configuration has changed.
+#
+
+try:
+    import os
+    import subprocess
+    import sys
+    import syslog
+    from swsssdk import ConfigDBConnector
+except ImportError as err:
+    raise ImportError("%s - required module not found" % str(err))
+
+VERSION = "1.0"
+
+SYSLOG_IDENTIFIER = "caclmgrd"
+
+
+# ========================== Syslog wrappers ==========================
+
+def log_info(msg):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+
+def log_warning(msg):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+
+def log_error(msg):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+
+# ============================== Classes ==============================
+
+class ControlPlaneAclManager(object):
+    """
+    Class which reads control plane ACL tables and rules from Config DB,
+    translates them into equivalent iptables commands and runs those
+    commands in order to apply the control plane ACLs.
+
+    Attributes:
+        config_db: Handle to Config Redis database via SwSS SDK
+    """
+    ACL_TABLE = "ACL_TABLE"
+    ACL_RULE = "ACL_RULE"
+
+    ACL_TABLE_TYPE_CTRLPLANE = "CTRLPLANE"
+
+    # To specify a port range, use iptables format: separate start and end
+    # ports with a colon, e.g., "1000:2000"
+    ACL_SERVICES = {
+        "NTP": {"ip_protocols": ["udp"], "dst_ports": ["123"]},
+        "SNMP": {"ip_protocols": ["tcp", "udp"], "dst_ports": ["161"]},
+        "SSH": {"ip_protocols": ["tcp"], "dst_ports": ["22"]}
+    }
+
+    def __init__(self):
+        # Open a handle to the Config database
+        self.config_db = ConfigDBConnector()
+        self.config_db.connect()
+
+    def run_commands(self, commands):
+        """
+        Given a list of shell commands, run them in order
+
+        Args:
+            commands: List of strings, each string is a shell command
+        """
+        for cmd in commands:
+            proc = subprocess.Popen(cmd, shell=True)
+
+            (stdout, stderr) = proc.communicate()
+
+            if proc.returncode != 0:
+                log_error("Error running command '{}'".format(cmd))
```

Comment:
Do we want to continue with the following commands when previous command failed? Not quite sure whether to stop or to continue is a better approach, might need to look into detail scenario.